src/category_theory/limits/limits.lean

```diff
@@ -7,20 +7,60 @@ import category_theory.adjunction.basic
 import category_theory.limits.cones
 import category_theory.reflects_isomorphisms
 
+/-!
+# Limits and colimits
+
+We set up the general theory of limits and colimits in a category.
+In this introduction we only describe the setup for limits;
+it is repeated, with slightly different names, for colimits.
+
+The three main structures involved are
+* `is_limit c`, for `c : cone F`, `F : J ⥤ C`, expressing that `c` is a limit cone,
+* `limit_cone F`, which consists of a choice of cone for `F` and the fact it is a limit cone, and
+* `has_limit F`, asserting the mere existence of some limit cone for `F`.
```

Comment:
This docstring morally depends on the prop-limits branch, right?

Reply:
err... Yes!

I realise also that I've now written a module doc for `limits/limits.lean` both on this branch, and on that branch! I guess when one gets merged, I'll merge conflicts and pick the best sentences from each. :-)